tests/unit/handle-changes-test.js

```diff
@@ -0,0 +1,44 @@
+var simple = require('simple-mock')
+var test = require('tape').test
+
+var handleChanges = require('../../lib/helpers/handle-changes')
+
+test('handleChanges changes during bootstrapping', function (t) {
+  t.plan(2)
+
+  var change = {
+    doc: {
+      foo: 'bar'
+    },
+    id: 'id1234'
+  }
+  var changesSpy = {
+    on: simple.spy().callFn(function (changeEvent, callback) {
+      callback(change)
+    })
+  }
+
+  var state = {
+    db: {
+      allDocs: simple.spy().resolveWith({ rows: [] }),
+      changes: simple.spy().returnWith(changesSpy)
+    },
+    emitter: {
+      emit: simple.spy()
+    }
+  }
+
+  handleChanges(state)
+
+  Promise.resolve(state.bootstrap)
```

Comment:
I'm not sure what you do here exactly :) `state.bootstrap` is undefined, so you this is basically a noop

What I meant in [my comment](https://github.com/hoodiehq/hoodie-store-client/pull/154#issuecomment-315632241) was to do something like this

```js
var resolveAllDocs
state.db.allDocs = new Promise(function (resolve) {
  resolveAllDocs = resolve
})
```

That way you can resolve `db.allDocs()` at any point you like. Now you can emit the events and then call `resolveAllDocs()`.

I think the current est works because `handleChanges` runs all synchronous, so the callback passed to `state.db.changes().on('change', callback)` gets executed immediately, while `simple.spy().resolveWith({ rows: [] })` resolves on next tick.